src/main/java/org/craftercms/engine/model/DefaultSiteItem.java

```diff
@@ -36,14 +40,14 @@
  */
 public class DefaultSiteItem extends AbstractXmlSiteItem {
 
-    protected Item item;
-    protected List<SiteItem> childItems;
-    protected Comparator<SiteItem> sortComparator;
+    private final ImmutableItem item;
```

Comment:
Please keep these fields protected just in case someone is extending this class in an overlay

Reply:
I fixed this to use `protected`. Also add methods `setItem`, `setSortComparator` to throw exceptions to prevent it from being set in Groovy.